src/task_change.rs

```diff
@@ -512,8 +594,11 @@ pub fn display_changeset(
     // Nice display
     if no_changes {
         println!("No changes.");
+        return;
```

Comment:
Any trivial amount of testing after adding such a new functionality would catch that, wouldn't it ?

Reply:
Well I'm not really sure, if eg. I add category A, and then test with both new tasks and a category A change, then I won't hit the failing code path (as the added new task would make `no_changes` be `false`).

And I must say I don't really see short-cutting 3 `if`s and a sort of an empty vector an advantage big enough to warrant this risk, however small it is, so long as Rust doesn't come with a formal proof framework strong enough to ensure that such a code path doesn't exist.

That said, upon more thought, maybe there'd be a way to avoid both unneeded computation and this code path:
1. set `no_changes = true`
2. inside each `if`, set `no_changes = false`
3. put this `if no_changes` test after all the `if`s

Reply:
I like this, and it turns out this is redundant with the `is_first` variable

Reply:
Oh indeed, I hadn't noticed :+1: